docs/src/routes/types.tsx

```diff
@@ -0,0 +1,101 @@
+import {Markdown} from '~/components/markdown/markdown';
+
+function types() {
+    const md = `
```

Comment:
This is missing the point of having markdown files that can be edited in github and previewed. Instead the text is hard to get right.

Reply:
Before shooting this down entirely, remember that the website has HMR (npm run dev) with a near-instant preview. It's excellent for editing.

Reply:
But then why use markdown? Simply code the relevant html/react here...

Reply:
The stack can handle a mix of both, but I think there in many cases are quite a lot more boilerplate when using html

Reply:
I initially took the .mdx approach with astro. Markdown with some JSX components. It worked for some things, but then we have all the code highlighting, which in markdown is indicated with a string. The problem is that we can't use template literals in Markdown, so having it this way we can easily inject variables in the markdown, and even the code snipped which MDX/MD doesn't allow for. That means that in fully markdown codebase, we would have to make a lot more (React) components for all the cases where Markdown fall short, and it's not pretty, so that's why I ended up in this approach.